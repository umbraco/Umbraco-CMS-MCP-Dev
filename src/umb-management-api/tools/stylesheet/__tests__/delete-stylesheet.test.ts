import DeleteStylesheetTool from "../delete/delete-stylesheet.js";
import { StylesheetHelper } from "./helpers/stylesheet-helper.js";
import { StylesheetBuilder } from "./helpers/stylesheet-builder.js";
import { jest } from "@jest/globals";

const TEST_STYLESHEET_NAME = "_TestDeleteStylesheet.css";
const TEST_CONTENT = "/* Test delete stylesheet */\nbody { color: red; }";
const NON_EXISTENT_STYLESHEET_PATH = "/_NonExistentStylesheet.css";

describe("delete-stylesheet", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    await StylesheetHelper.cleanup(TEST_STYLESHEET_NAME);
    console.error = originalConsoleError;
  });

  it("should delete a stylesheet", async () => {
    // Arrange - Create a stylesheet first using builder
    const stylesheetBuilder = new StylesheetBuilder()
      .withName(TEST_STYLESHEET_NAME)
      .withContent(TEST_CONTENT);
    
    await stylesheetBuilder.create();
    const stylesheetPath = stylesheetBuilder.getPath();

    // Verify it exists before deletion
    const existsBefore = await StylesheetHelper.verifyStylesheet(stylesheetPath);
    expect(existsBefore).toBe(true);

    // Act
    const result = await DeleteStylesheetTool().handler({ path: stylesheetPath }, { signal: new AbortController().signal });

    // Assert
    expect(result).toMatchSnapshot();

    // Verify the stylesheet no longer exists
    const existsAfter = await StylesheetHelper.verifyStylesheet(stylesheetPath);
    expect(existsAfter).toBe(false);
  });

  it("should handle non-existent stylesheet", async () => {
    // Act
    const result = await DeleteStylesheetTool().handler({ path: NON_EXISTENT_STYLESHEET_PATH }, { signal: new AbortController().signal });

    // Assert
    expect(result).toMatchSnapshot();
  });
});