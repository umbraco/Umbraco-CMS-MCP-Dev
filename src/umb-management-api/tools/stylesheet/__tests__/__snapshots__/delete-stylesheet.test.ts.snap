// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete-stylesheet should delete a stylesheet 1`] = `
{
  "content": [
    {
      "text": """",
      "type": "text",
    },
  ],
}
`;

exports[`delete-stylesheet should handle non-existent stylesheet 1`] = `
{
  "content": [
    {
      "text": "Error using delete-stylesheet:
{
  "message": "Request failed with status code 404",
  "response": {
    "type": "Error",
    "title": "Stylesheet not found",
    "status": 404,
    "detail": "The stylesheet was not found.",
    "operationStatus": "NotFound"
  }
}",
      "type": "text",
    },
  ],
}
`;
